CunnecryConverterBuildSprint1/CunnecryConverterBuildSprint1/ConverterViewController.swift

```diff
@@ -0,0 +1,115 @@
+//
+//  ConverterViewController.swift
+//  CunnecryConverterBuildSprint1
+//
+//  Created by Dojo on 6/16/20.
+//  Copyright © 2020 Dojo. All rights reserved.
+//
+
+import UIKit
+
+class ConverterViewController: UIViewController, UIPickerViewDataSource, UIPickerViewDelegate {
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        currencyController.fetchRates()
+        currencyController.delegate = self
+    }
+    
+    @IBOutlet weak var currencyPicker: UIPickerView!
+    
+    
+    @IBOutlet weak var amountToConvertTextField: UITextField!
+    
+    
+    @IBOutlet weak var convertedAmountTextField: UITextField!
+    
+    @IBOutlet weak var currencyDisplay: UILabel!
+    
+    let currencyController = CurrencyController()
+    
+    var newRates: [String: Double] = [:]
+    
+    let currencyNames = ["CAD, Canadian Dollar",
+                         "HKD, Hong Kong Dollar",
+                         "ISK, Icelandic Króna",
+                         "PHP, Philippine peso",
+                         "DKK, Danish Krone",
+                         "HUF, Hungarian Forint",
+                         "CZK, Czech Koruna",
+                         "GBP, Pound sterling",
+                         "RON, Romanian Leu",
+                         "SEK, Swedish Krona",
+                         "IDR, Indonesian Rupiah",
+                         "INR, Indian Rupee",
+                         "BRL, Brazilian Real",
+                         "RUB, Russian Ruble",
+                         "HRK, Croatian Kuna",
+                         "JPY, Japanese Yen",
+                         "THB, Thailand Baht",
+                         "CHF, Swiss Franc",
+                         "EUR, Euro",
+                         "MYR, Malaysian Ringgit",
+                         "BGN, Bulgarian Lev",
+                         "TRY, Turkish lira",
+                         "CNY, Chinese Yuan",
+                         "NOK, Norwegian Krone",
+                         "NZD, New Zealand Dollar",
+                         "ZAR, South African Rand",
+                         "USD, United States Dollar",
+                         "MXN, Mexican Peso",
+                         "SGD, Singapore Dollar",
+                         "AUD, Australian Dollar",
+                         "ILS, Israeli New Shekel",
+                         "KRW, South Korean Won",
+                         "PLN, Poland złoty"]
+    
+    
+    func pickerView(_ pickerView: UIPickerView, numberOfRowsInComponent component: Int) -> Int {
+        return newRates.count
+    }
+    
+    func numberOfComponents(in pickerView: UIPickerView) -> Int {
+        1
+    }
+    
+    func pickerView(_ pickerView: UIPickerView, titleForRow row: Int, forComponent component: Int) -> String? {
+        return "\(currencyNames.sorted()[row])"
```

Comment:
Notice that your number of rows on line 69 uses newRates.count and the titleForRow here uses the currencyNames array. What do you think would happen if the newRates had more rates than currencyNames had names? (Hint: it would make your app crash)

Reply:
it's funny that you mention this.  A few days after a commit i got back to the project and it was crashing, turns out they added Thai baht to the API... (or maybe i left it out of the names dictionary by mistake 🤷🏻‍♂️).  I fixed it by adding in the missing currency but guess i forgot to change the .count, i did immediately after you pointed this out 😁